lib/media/track/localvideotrack.js

```diff
@@ -102,7 +130,18 @@ class LocalVideoTrack extends LocalMediaVideoTrack {
    * });
    */
   restart() {
-    return super.restart.apply(this, arguments);
+    if (this._workaroundSilentLocalVideoCleanup) {
+      this._workaroundSilentLocalVideoCleanup();
+      this._workaroundSilentLocalVideoCleanup = null;
+    }
+    const promise = super.restart.apply(this, arguments);
+
+    if (this._workaroundSilentLocalVideo) {
+      promise.finally(() => {
```

Comment:
I think it's better to make caller wait for cleanup
```
      promise = promise.finally(() => {
        this._workaroundSilentLocalVideoCleanup = this._workaroundSilentLocalVideo(this, document);
      });
```

Reply:
@makarandp0 The problem with this is that any error thrown in earlier promise chains may not be passed to the Promise returned by `finally()`. Also, I don't think any application code needs to depend on the workaround function being called.